Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -329,6 +330,49 @@ extension BrowserViewController: URLBarDelegate {
         hideSearchController()
         updateInContentHomePanel(tabManager.selectedTab?.url)
     }
+
+    func urlBar(urlBar: URLBarView, suggestionForText text: String, complete: String? -> Void) {
+        let options = QueryOptions(filter: text, filterType: FilterType.Url, sort: QuerySort.Frecency)
+        profile.history.get(options, complete: { (cursor: Cursor) in
```

Comment:
Redoing a second query on the db for every keypress seems pretty wasteful. Can we reuse the cursor we have?